lib/bundler/fetcher/compact_index.rb

```diff
@@ -122,14 +122,24 @@ def call(path, headers)
       end
 
       def md5_available?
+        return true unless fips_enabled? && Process.respond_to?(:fork)
+        pid = fork do
```

Comment:
- Platforms that aren't fork-safe will not reach this line of code because `Process.respond_to?(:fork)` will be `false`.
- Performance-wise, I expect a process fork to perform better than disabling the compact-index entirely, so for the case where FIPS mode is available but not active this should be a performance improvement. The tradeoff is that this causes the case where FIPS mode is active to perform  the fork.
- As far as I can tell, there is no way via Ruby's stdlib to determine in-process if the MD5 digest is available. The process is aborted as soon as OpenSSH's MD5 module is invoked, so in order to detect this failure without aborting the parent process, it's necessary to perform the test on a child process.